RK-coeff-opt/nonlinear_constraints.m

```diff
@@ -120,3 +120,66 @@
     res = polyval(poly_coef, constrain_emb_stability);
     con = [con, res .* conj(res) - 1];
 end
+
+q=1;
+% Throw error message in case q>7
+msg = 'The energy-preserving conditions for orders q>7 have not been implemented.';
+% PEP conditions
+if q >= 8
+    error(msg);
+end
```

Comment:
I would put this at the very end of the `if` blocks below (since people will look there if they add higher-order constraints).